src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -33,31 +42,61 @@ export class EditorComponent implements OnInit {
     });
 
     this.ctx = this.canvas.nativeElement.getContext('2d');
+    this.hiddenCtx = this.hiddenCanvas.nativeElement.getContext('2d');
     this.draw();
-  }  
+  }   
   
-  //  Checks that route for editor URL is in bounds of array of Urls and is a number
-  //  Redirects to gallery component if not
-  private checkImageUrl(index: number) {
+  /**
+   * Checks that route for editor URL is in bounds of array of Urls and is a number
+   * Redirects to gallery component if not, does not complete editor component
+   * index: the route from the editor url passed through gallery.ts 
+   */
+  private checkImageUrl(index: number): void {
     if (Number.isNaN(index) || index >= imageUrls.length || index < 0) {    
       console.log('Index: ' + index + 'Not a number or greater than number of images, returning to Gallery');
       this.router.navigate(['/gallery']);
     }
   }
-  
-  // Draws the image user selects from gallery on Canvas
-  // Currently, editor can require user to scroll to access
-  //  the entire photo, need to make it so the editor is 
-  //  fixed to screen size
-  private draw() {
+  /**  
+   * Draws the image user selects from gallery on Canvas
+   * Currently, editor can require user to scroll to access
+   * the entire photo, need to make it so the editor is 
+   * fixed to screen size
+   */
+  private draw(): void {
     this.image.src = this.url;
 
-    //adjust canvas to image width and height
-    this.canvas.nativeElement.width = this.image.width;
-    this.canvas.nativeElement.height = this.image.height;
+    //init hidden canvas width to original images width and height
+    this.hiddenCanvas.nativeElement.width = this.image.width;
+    this.hiddenCanvas.nativeElement.height = this.image.height;
+
+    // used to scale the image to the window size, 1.2 so the image is smaller than the window.
+    this.scaleFactor = Math.floor(this.innerHeight / this.image.height / this.scaleFactor);
```

Comment:
It might be more intuitive to have a scale factor < 1 to mean the image is smaller than the window. So instead of dividing by 1.2, multiplying by 0.83?